### PR TITLE
Add --device support for Windows

### PR DESCRIPTION
**Pull request by jterry75**
Adds support for --device in Windows. This must take the form of:
--device='class/{clsid}'

Signed-off-by: Justin Terry (VM) <juterry@microsoft.com>

@jhowardmsft - FYI

### CODE REVIEW

**Comment by jterry75** [comment]
See: [moby/moby:37638](https://github.com/moby/moby/pull/37638) for daemon side PR

File: cli/command/container/opts_windows.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+// validateDevice validates a path for devices
+// It will make sure 'val' is in the form:
+//    class/{clsid}
+func validateDevice(val string) (string, error) {
+	arr := strings.Split(val, ":")
+	switch len(arr) {
+	case 1:
+		if !strings.HasPrefix(arr[0], "class") {
+			return "", errors.New("device just have prefix: 'class'")
```
Comment (jhowardmsft) [inline cli/command/container/opts_windows.go]: Must, not just?

**Comment by jhowardmsft** [comment]
Think there's a typo on an error string. Otherwise code LGTM (not a maintainer in this repo)

**Comment by jterry75** [comment]
@theJeztah - Is there anything I can help with to get this in?

**Comment by jterry75** [comment]
@jhowardmsft - This is the client side. PTAL

**Comment by jhowardmsft** [comment]
LGTM (not a maintainer)

**Comment by jhowardmsft** [comment]
@thaJeztah Can this be merged now the daemon side is in?

File: cli/command/container/opts_unix.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
+// +build !windows
```
[the remark this answers is not in the document]
Comment (jterry75) [reply cli/command/container/opts_unix.go]: How do you suggest I determine when to use what parser for the device assignment? I would have to ping the end node to determine if its a Windows or Linux host first?

[the remark this answers is not in the document]
Comment (jhowardmsft) [reply cli/command/container/opts_unix.go]: Take a look at cli\command\cli.go L212, function initializeFromClient. You can pull the server OS out of `cli.serverInfo.OSType`.  IIRC every CLI command starts with a _ping API.

[the remark this answers is not in the document]
Comment (thaJeztah) [reply cli/command/container/opts_unix.go]: Yes, unfortunately it gets complicated fast to take mixed environments into account 😞. I guess we should keep the separate functions for windows and linux (to keep it clean), and add a wrapper function that calls one of them, based on what the target platform is

[the remark this answers is not in the document]
Comment (thaJeztah) [reply cli/command/container/opts_unix.go]: If _possible_ (not sure that is in this case (haven't looked in depth)), we try to reduce validation on the client side to a minimum, and leave it to the daemon to return an error.

**Comment by jterry75** [comment]
Closing in favor of #1606

**Pull request closed without merging**